File: AdventureWorks/AdventureWorks.Client.Spa/DataObjects/Sales/SalesOrderCustomerObjectCustomized.ts

import { SalesOrderCustomerObject as GeneratedDataObject } from 'DataObjects/Sales/SalesOrderCustomerObject';
import { BusinessEntityAddressReadListCacheLoader } from '../../CacheLoaders/BusinessEntityAddressReadListCacheLoader';
import { BusinessEntityAddress } from '../../Enumerations/Enumerations';

export class SalesOrderCustomerObject extends GeneratedDataObject {

    private addressLoader: BusinessEntityAddressReadListCacheLoader;

    onInitialized() {
        this.LookupObject.TrackModifications = false;

        this.addressLoader = new BusinessEntityAddressReadListCacheLoader();
        this.BillingAddressObject.AddressId.LocalCacheLoader = this.addressLoader;
        this.ShippingAddressObject.AddressId.LocalCacheLoader = this.addressLoader;

        this.StoreId.InternalValue.subscribe(this.onCustomerChanged, this);
        this.PersonId.InternalValue.subscribe(this.onCustomerChanged, this);

        super.onInitialized();
    }

    onCustomerChanged() {
        if (this.PersonId.isNull() && this.StoreId.isNull()) return;

        var newParams = {};
        newParams[BusinessEntityAddress.Parameters.BusinessEntityId] = this.StoreId.isNull() ?
            this.PersonId.TransportValue() : this.StoreId.TransportValue();
        this.addressLoader.setParameters(newParams, () => {
            this.BillingAddressObject.AddressId.clearInvalidValues();
            this.ShippingAddressObject.AddressId.clearInvalidValues();
            this.BillingAddressObject.AddressId.updateValueList();
            this.ShippingAddressObject.AddressId.updateValueList();
        });
    }
}

File: AdventureWorks/AdventureWorks.Client.Common/DataObjects/Sales/SalesOrderCustomerObjectCustomized.cs
using AdventureWorks.Services;
using System;
using System.Collections.Generic;
using Xomega.Framework;
using Xomega.Framework.Lookup;

namespace AdventureWorks.Client.Objects
{
    public class SalesOrderCustomerObjectCustomized : SalesOrderCustomerObject
    {
        public SalesOrderCustomerObjectCustomized()
        {
        }

        public SalesOrderCustomerObjectCustomized(IServiceProvider serviceProvider) : base(serviceProvider)
        {
        }

        // construct properties and child objects
        protected override void Initialize()
        {
            base.Initialize();
        }

        private LocalLookupCacheLoader AddressLoader;

        // perform post intialization
        protected override void OnInitialized()
        {
            base.OnInitialized();

            AddressLoader = new BusinessEntityAddressReadListCacheLoader(ServiceProvider);
            BillingAddressObject.AddressIdProperty.LocalCacheLoader = AddressLoader;
            ShippingAddressObject.AddressIdProperty.LocalCacheLoader = AddressLoader;

            StoreIdProperty.Change += OnCustomerChanged;
            PersonIdProperty.Change += OnCustomerChanged;
        }

        private void OnCustomerChanged(object sender, PropertyChangeEventArgs e)
        {
            if (!e.Change.IncludesValue() || Equals(e.OldValue, e.NewValue) ||
                PersonIdProperty.Value == null && StoreIdProperty.Value == null) return;

            int entityId = StoreIdProperty.Value == null ? // use store or person id
                PersonIdProperty.Value.Value : StoreIdProperty.Value.Value;

            AddressLoader.SetParameters(new Dictionary<string, object>() {
                { Enumerations.BusinessEntityAddress.Parameters.BusinessEntityId, entityId }
            });

            BillingAddressObject.AddressIdProperty.ClearInvalidValues();
            ShippingAddressObject.AddressIdProperty.ClearInvalidValues();

            var args = new PropertyChangeEventArgs(PropertyChange.Items, null, null, e.Row);
            BillingAddressObject.AddressIdProperty.FirePropertyChange(args);
            ShippingAddressObject.AddressIdProperty.FirePropertyChange(args);
        }
    }
}